### ISSUE TITLE
[Feature Request] Add option to save header when saving MSG file

### ISSUE THREAD

**Issue by karpaz** [TeamMsgExtractor/msg-extractor#198]
Hello everyone,
It's not a bug, it's a feature request, or maybe help to someone else.

I want to have in output also header from original *.msg file. Is it possible, or should I use any flag when converting the message?

**Comment by TheElementalOfDestruction** [comment]
Please avoid using an i descriptive title like "feature request". The problem with doing this is it can make it harder to find related issues if the titling is poor. 

Moving on, I'm not sure I understand what you are requesting, but I think you are asking for the ability to either have the header saved as it's own file or have all of it's fields saved into a file. Please let me know if I got that correct or am mistaken.

If that is not what you were asking for, you can likely disregard this next part, but if it is then the nature of .msg files makes this difficult. The header field is considered *optional* so telling the program to save it would be rather hit or miss. I could have the program save whatever it *does* find, but there is no guarantee it will actually be what you want. Sorry, but there's really nothing I can do about that.

**Comment by karpaz** [comment]
First, I'd like to apologize for using the confusing "feature request". What you mentioned in the second paragraph is true, my request is to have the header either stored in its own file or in the message content file. The main reason for this request is that I am working on a Linux OS and I need to perform header analysis as well. If you can't do anything about this, I apologize and please close this ticket. Have a nice day.

**Comment by TheElementalOfDestruction** [comment]
I'll add an option to, if it is found, save the raw header to a file in the next update (The major save rewrite). What you can do for now is try to take a look at the python code for this project and create your own main function. Alternatively, you can use the current main function but also call a secondary script to save the header to a file that has code something like this:
```python
import extract_msg

# This part should work, but if it doesn't you can just switch it to manually assigning to a variable and manually closing it afterwards.
with extract_msg.openMsg('path/to/msgfile.msg') as msg: # Open the msg file.
    headerText = self._getStringStream('__substg1.0_007D') # Get the text for the header.
    if headerText: # Does the header actually exist?
        with open('path/to/save/location/header.txt', 'wb') as f: # Open a file to save the header text into.
            f.write(headerText.encode('utf-8') # Save the header text to a file.
```
You can modify this code to take in a path from the user in some way, or even take in the raw msg data if needed. This code will give you the raw, unprocessed header. This code will also only save the header if it can find it. Unfortunately, because headers are optional, this may frequently save nothing.

Thank you for the clarification. I'm not upset, just thought this a good time to provide a learning opportunity. Now that I understand what you are looking for, I'll adjust the title to best match the request.

**Comment by TheElementalOfDestruction** [comment]
Whoops, this was supposed to be implemented in 0.29.0 but I kind of forgot to do it, so it has a mention of it inside the save documentation but it won't do anything. I'll put a quick fix for that in 0.29.1

**Comment by TheElementalOfDestruction** [comment]
Okay this should be implemented now. Let me know if you have any issues.
